internal/proxy/look_aside_balancer.go

```diff
@@ -0,0 +1,185 @@
+// Licensed to the LF AI & Data foundation under one
+// or more contributor license agreements. See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership. The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package proxy
+
+import (
+	"context"
+	"math"
+	"sync"
+	"time"
+
+	"github.com/milvus-io/milvus-proto/go-api/v2/commonpb"
+	"github.com/milvus-io/milvus/internal/proto/internalpb"
+	"github.com/milvus-io/milvus/pkg/log"
+	"github.com/milvus-io/milvus/pkg/util/merr"
+	"github.com/milvus-io/milvus/pkg/util/typeutil"
+	"go.uber.org/atomic"
+	"go.uber.org/zap"
+)
+
+var (
+	checkQueryNodeHealthInterval = 500 * time.Millisecond
+)
+
+type LookAsideBalancer struct {
+	clientMgr shardClientMgr
+
+	// query node -> workload latest metrics
+	metricsMap *typeutil.ConcurrentMap[int64, *internalpb.CostAggregation]
+
+	// query node -> last update metrics ts
+	metricsUpdateTs *typeutil.ConcurrentMap[int64, int64]
+
+	// query node -> total nq of requests which already send but response hasn't received
+	executingTaskTotalNQ *typeutil.ConcurrentMap[int64, *atomic.Int64]
```

Comment:
Why does concurrentmap need to store an atomic.Int64 instead of int64?

Reply:
cause we need to modify the value in `SelectNode`, if we use int64 here,  during concurrent case, it will set a wrong value back during modify